api/api_gateway/api.py

```diff
@@ -851,3 +852,73 @@ def email_list_import(
         return {"error": f"error importing list: {error}"}
     else:
         return {"status": "OK"}
+
+
+class ListImportPayload(BaseModel):
+    email: Optional[conlist(EmailStr, min_items=1, max_items=10000)]
+    phone: Optional[
+        conlist(
+            constr(
+                strip_whitespace=True,
+                min_length=9,
+                max_length=15,
+            ),
+            min_items=1,
+            max_items=10000,
+        )
+    ]
+
+    class Config:
+        extra = "forbid"
+
+
+@app.post("/list/{list_id}/import")
+def list_import(
+    list_id,
+    list_import_payload: ListImportPayload,
+    response: Response,
+    session: Session = Depends(get_db),
+    _authorized: bool = Depends(verify_token),
+):
+    """Imports a list"""
+    try:
+        _ = session.query(List).filter(List.id == list_id).one()
+
+        if not list_import_payload.email and not list_import_payload.phone:
+            response.status_code = status.HTTP_422_UNPROCESSABLE_ENTITY
+            return {"error": "Payload must include one of: phone<list>, email<list>"}
+
+        if list_import_payload.email and list_import_payload.phone:
+            response.status_code = status.HTTP_422_UNPROCESSABLE_ENTITY
+            return {
+                "error": "Payload may only include one of: phone<list>, email<list>"
+            }
+
+        if list_import_payload.email:
+            session.add_all(
+                [
+                    Subscription(email=email, confirmed=True, list_id=list_id)
+                    for email in list_import_payload.email
+                ]
+            )
+            session.commit()
+
+        if list_import_payload.phone:
+            session.add_all(
+                [
+                    Subscription(phone=number, confirmed=True, list_id=list_id)
+                    for number in list_import_payload.phone
+                ]
+            )
+            session.commit()
+
+    except NoResultFound:
+        response.status_code = status.HTTP_404_NOT_FOUND
+        return {"error": "list not found"}
+    except SQLAlchemyError as error:
+        metrics.add_metric(name="ListEmailImportError", unit=MetricUnit.Count, value=1)
+        metrics.add_metadata(key="list_id", value=str(list_import_payload.list_id))
+        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+        return {"error": f"error importing list: {error}"}
```

Comment:
Might want to add the "type" of import here

Reply:
👍